bin/tx-prover/src/proxy/mod.rs

```diff
@@ -284,25 +298,20 @@ impl ProxyHttp for LoadBalancer {
         e
     }
 
-    /// Logs the request lifecycle in case that an error happened and removes the request from the
-    /// worker queue.
+    /// Logs the request lifecycle in case that an error happened and sets the worker as available.
     ///
     /// This method is the last one in the request lifecycle, no matter if the request was
     /// processed or not.
-    async fn logging(&self, session: &mut Session, e: Option<&Error>, _ctx: &mut Self::CTX)
+    async fn logging(&self, _session: &mut Session, e: Option<&Error>, ctx: &mut Self::CTX)
     where
         Self::CTX: Send + Sync,
     {
         if let Some(e) = e {
             error!("Error: {:?}", e);
         }
 
-        // Get the request ID from the session
-        let request_id = Self::get_request_id(session).expect("Request ID not found");
-
-        // Remove the completed request from the worker queue
-        // Maybe we can replace this with a read lock and using write only in the moment of the
-        // deletion.
-        Self::remove_request_from_queue(request_id).await;
+        // Mark the worker as available
+        self.add_available_worker(ctx.worker.take().expect("Failed to get worker"))
```

Comment:
I think I'd add an `info!()` log here (or maybe better within the `add_available_worker` function) that mentions that the worker became available. Basically I'm trying to think if we have enough request and traceability to be able to tell how the system was working if something went wrong.

Reply:
Adding it.